app/controllers/teams_controller.rb

```diff
@@ -46,12 +47,34 @@ def destroy
     redirect_to teams_url, notice: 'Team was successfully destroyed.', status: :see_other
   end
 
+  def announcements
+    @q = @team.announcements.ransack(params[:q])
+    @announcements = @q.result(distinct: true).page(params[:page])
+    render 'announcements/index'
+  end
+
+  def badges
+    @q = @team.badges.ransack(params[:q])
+    @badges = @q.result(distinct: true).page(params[:page])
+    render 'badges/index'
+  end
+
+  def events
+    @q = @team.events.ransack(params[:q])
+    @events = @q.result(distinct: true).page(params[:page]).includes(:event_type)
+    render 'events/index'
+  end
+
```

Comment:
You should leverage the ransack search functionality built in to the existing index pages and link to those with the appropriate parameter set to filter down to the stuff for the current team rather than duplicating that functionality here.